### PR TITLE
[15.0][IMP] resource_booking: Add multiples requesters in booking

### PR DESCRIPTION
**Pull request by carolinafernandez-tecnativa**
- Add multiples requesters in booking

@Tecnativa
TT45338

@pedrobaeza

### CODE REVIEW

File: resource_booking/models/resource_booking.py [hunk 1]
```diff
@@ -125,7 +125,12 @@ class ResourceBooking(models.Model):
     description = fields.Html()
     partner_id = fields.Many2one(
         comodel_name="res.partner",
+        compute="_compute_partner_id",
         string="Requester",
+    )
+    partner_ids = fields.Many2many(
+        comodel_name="res.partner",
+        string="Requesters",
         index=True,
```
Comment (pedrobaeza) [inline resource_booking/models/resource_booking.py]: `index` doesn't make sense in m2m, nor `ondelete`.

File: resource_booking/models/resource_booking.py [hunk 2]
```diff
@@ -674,11 +686,11 @@ def action_confirm(self):
                 if not booking.meeting_id:
                     continue
                 # Make sure requester and user resources are meeting attendees
```
Comment (pedrobaeza) [inline resource_booking/models/resource_booking.py]:
```suggestion
                # Make sure requesters and user resources are meeting attendees
```

File: resource_booking/models/resource_booking.py (hunk 1 above)
Comment (carolinafernandez-tecnativa) [reply resource_booking/models/resource_booking.py]: remove index=True. Should i keep ondelete?

Comment (pedrobaeza) [reply resource_booking/models/resource_booking.py]: Please remove it

**Comment by pedrobaeza** [comment]
This is not ADD, but IMP, and the commit message shouldn't contain the version.

**Comment by carolinafernandez-tecnativa** [comment]
> This is not ADD, but IMP, and the commit message shouldn't contain the version.
Done, thanks!

File: migrations/15.0.1.4.0/post-migration.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Copyright 2023 Tecnativa - Carolina Fernandez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from psycopg2.extensions import AsIs
+
+from odoo import SUPERUSER_ID, api
+
+
+def migrate(cr, version):
+    """Put partner_id in partner_ids"""
+    env = api.Environment(cr, SUPERUSER_ID, {})
```
Comment (pedrobaeza) [inline migrations/15.0.1.4.0/post-migration.py]: Use `openupgradelib.m2o_to_x2x` method instead.

Comment (carolinafernandez-tecnativa) [reply migrations/15.0.1.4.0/post-migration.py]: could you please check it now? Thanks!

File: migrations/15.0.1.4.0/pre-migration.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Copyright 2023 Tecnativa - Carolina Fernandez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+def migrate(env):
+    env.cr.execute(
+        """
+        ALTER TABLE resource_booking RENAME partner_id TO migrated_partner_id
```
Comment (pedrobaeza) [inline migrations/15.0.1.4.0/pre-migration.py]: Better to use a familiar nomenclature (and more expressive):

```suggestion
        ALTER TABLE resource_booking RENAME partner_id TO old_partner_id
```

**Review by pedrobaeza — COMMENTED** [review]
I'm thinking that now that we put several partners, we should switch the nomenclature from "Requester" to "Attendees". What do you think?

File: migrations/15.0.1.4.0/pre-migration.py (hunk 4 above)
Comment (pedrobaeza) [reply migrations/15.0.1.4.0/pre-migration.py]: And technically FYI, there's no need to rename the column, as this column is available on post-migration.

Comment (carolinafernandez-tecnativa) [reply migrations/15.0.1.4.0/pre-migration.py]: Ok,  so if the column is available i deleted the file pre-migration.py

**Comment by carolinafernandez-tecnativa** [comment]
> I'm thinking that now that we put several partners, we should switch the nomenclature from "Requester" to "Attendees". What do you think?

I agree. I just changed :)

File: migrations/15.0.1.4.0/pre-migration.py (hunk 4 above)
Comment (pedrobaeza) [reply migrations/15.0.1.4.0/pre-migration.py]: Oh, wait, I didn't see that you are still defining `partner_id` as a regular field, but not stored, so in this case the column is removed, so please rename. Sorry for not seeing before that issue.

**Review by pedrobaeza — APPROVED** [review]
Thanks for the new feature:

/ocabot merge nobump

**Comment by pedrobaeza** [comment]
Please review 16.0 migration and then let's include this one after the merge.

**Pull request merged**